web/src/components/applications-page/encrypt-secret-drawer/index.tsx

```diff
@@ -0,0 +1,216 @@
+
+export interface EncryptSecretDrawerProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+const DRAWER_TITLE = "Encrypting secret data";
+const BASE64_CHECKBOX_LABEL =
+  "Use base64 encoding before encrypting the secret";
+
+const validationSchema = yup.object({
+  secretData: yup.string().required(),
+  base64: yup.bool(),
+  pipedId: yup.string().required("Please select a piped"),
+});
+
+export const EncryptSecretDrawer: FC<EncryptSecretDrawerProps> = memo(
+  function EncryptSecretDrawer({ open, onClose }) {
+    const dispatch = useAppDispatch();
+
+    const pipeds = useAppSelector<Piped.AsObject[]>(selectAllPipeds);
+    const sealedSecret = useAppSelector<string | null>(
+      (state) => state.sealedSecret.data
+    );
+
+    const pipedOptions = useMemo(() => {
+      return pipeds
+        .filter((piped) => !piped.disabled)
+        .sort((a, b) => sortFunc(a.name, b.name));
+    }, [pipeds]);
+
+    useEffect(() => {
+      if (open) {
+        dispatch(fetchPipeds(false));
+      }
+    }, [dispatch, open]);
+
+    const formik = useFormik({
+      initialValues: {
+        secretData: "",
+        base64: false,
+        pipedId: "",
+      },
+      validationSchema,
+      async onSubmit(values) {
+        await dispatch(
+          generateSealedSecret({
+            data: values.secretData,
+            pipedId: values.pipedId,
+            base64Encoding: values.base64,
+          })
+        );
+      },
+    });
+
+    const handleClose = useCallback(() => {
+      onClose();
+      dispatch(clearSealedSecret());
+    }, [dispatch, onClose]);
+
+    const prevOpen = useRef(false);
+    useEffect(() => {
+      if (!prevOpen.current && open) {
+        formik.resetForm();
+      }
+      prevOpen.current = open;
+    }, [open, formik]);
+
+    return (
+      <Drawer
+        anchor="right"
+        open={open}
+        onClose={handleClose}
+        sx={{
+          "& .MuiDrawer-paper": {
+            width: 600,
+          },
+        }}
+      >
+        <Box sx={{ p: 3 }}>
+          <Typography variant="h6" sx={{ mb: 3 }}>
+            {DRAWER_TITLE}
+          </Typography>
+
+          {sealedSecret ? (
+            <Box>
+              <Typography variant="caption" color="textSecondary">
+                Encrypted secret data
+              </Typography>
+              <TextWithCopyButton
+                name="Encrypted secret"
+                value={sealedSecret}
+              />
+              <Box sx={{ mt: 3, display: "flex", justifyContent: "flex-end" }}>
+                <Button onClick={handleClose}>Close</Button>
+              </Box>
+            </Box>
+          ) : (
+            <form onSubmit={formik.handleSubmit}>
+              <FormControl
+                fullWidth
+                margin="dense"
+                error={formik.touched.pipedId && Boolean(formik.errors.pipedId)}
+              >
+                <InputLabel id="piped-select">Piped</InputLabel>
```

Comment:
This InputLabel should have require => so UI will appear "Piped *", it will make UI consistency when emphasize which input which required.

```suggestion
                <InputLabel id="piped-select" required>Piped</InputLabel>
```